src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/MigrateAlertIntTest.kt

```diff
@@ -0,0 +1,293 @@
+  }
+
+  @Test
+  fun `405 method not allowed`() {
+    val response = webTestClient.patch()
+      .uri("/migrate/alerts")
+      .headers(setAuthorisation(roles = listOf(ROLE_ALERTS_WRITER)))
+      .exchange()
+      .expectStatus().isEqualTo(HttpStatus.METHOD_NOT_ALLOWED)
+      .expectBody(ErrorResponse::class.java)
+      .returnResult().responseBody
+
+    with(response!!) {
+      assertThat(status).isEqualTo(405)
+      assertThat(errorCode).isNull()
+      assertThat(userMessage).isEqualTo("Method not allowed failure: Request method 'PATCH' is not supported")
+      assertThat(developerMessage).isEqualTo("Request method 'PATCH' is not supported")
+      assertThat(moreInfo).isNull()
+    }
+  }
+
+  @Test
+  fun `should return populated alert model`() {
+    val request = migrateAlertRequest()
+
+    val alert = webTestClient.migrateAlert(request = request)
+    val expectedCreatedTime = LocalDateTime.of(request.createdAt.year, request.createdAt.month, request.createdAt.dayOfMonth, request.createdAt.hour, request.createdAt.minute, request.createdAt.second)
+    assertThat(alert).isEqualTo(
+      AlertModel(
+        alert.alertUuid,
+        request.prisonNumber,
+        alertCodeVictimSummary(),
+        request.description,
+        request.authorisedBy,
+        request.activeFrom,
+        request.activeTo,
+        true,
+        emptyList(),
+        expectedCreatedTime,
+        request.createdBy,
+        request.createdByDisplayName,
+        null,
+        null,
+        null,
+      ),
+    )
+  }
+
+  @Test
+  fun `should migrate new alert`() {
+    val request = migrateAlertRequest()
+
+    val alert = webTestClient.migrateAlert(request = request)
+
+    val alertEntity = alertRepository.findByAlertUuid(alert.alertUuid)!!
+    val alertCode = alertCodeRepository.findByCode(request.alertCode)!!
+
+    assertThat(alertEntity).usingRecursiveAssertion().ignoringFields("auditEvents").isEqualTo(
+      Alert(
+        1,
+        alert.alertUuid,
+        alertCode,
+        request.prisonNumber,
+        request.description,
+        request.authorisedBy,
+        request.activeFrom!!,
+        request.activeTo,
+        alertEntity.migratedAt,
+      ),
+    )
+    with(alertEntity.auditEvents().single()) {
+      assertThat(auditEventId).isEqualTo(1)
+      assertThat(action).isEqualTo(AuditEventAction.CREATED)
+      assertThat(description).isEqualTo("Migrated alert created")
+      assertThat(actionedAt).isEqualToIgnoringNanos(request.createdAt)
+      assertThat(actionedBy).isEqualTo("AG111QD")
+      assertThat(actionedByDisplayName).isEqualTo("A Creator")
+    }
+
+    assertThat(alertEntity.migratedAt).isCloseToUtcNow(within(3, ChronoUnit.SECONDS))
+  }
+
+  @Test
+  fun `should not publish alert created event`() {
+    val request = migrateAlertRequest()
+
+    webTestClient.migrateAlert(request = request)
+    verify(hmppsQueueService, never()).findByTopicId(any())
```

Comment:
Unfortunately I think you'll have to put an artificial Thread.Sleep(1000) to be confident of this test assertion I think